java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ObjectTranslation.java

```diff
@@ -273,5 +345,16 @@ public static ObjectTranslation createThreadTranslation(JPDADebuggerImpl debugge
     public static ObjectTranslation createLocalsTranslation(JPDADebuggerImpl debugger) {
         return new ObjectTranslation(debugger, LOCALS_ID);
     }
-    
+
+    private static final class TranslationFuture extends CompletableFuture<Object> {
+
+        @Override
+        public Object get() {
+            try {
+                return super.get();
+            } catch (ExecutionException | InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)